pkg/system/phase2_creating.go

```diff
@@ -133,3 +136,70 @@ func (r *Reconciler) SetDesiredCoreApp() {
 		// r.Own(pvc)
 	}
 }
+
+// EnsureCredentialsRequestAndGetBucketName creates a CredentialsRequest resource if neccesary and returns
+// the bucket name allowed for the credentials. nil is returned if cloud credentials are not supported
+func (r *Reconciler) EnsureCredentialsRequestAndGetBucketName(log *logrus.Entry) error {
```

Comment:
I suggest a simpler name, and also seems like you don't need to pass the logger:
```suggestion
func (r *Reconciler) ReconcileCredentialsRequest() error {
```